addon/utils/plugins/lists/delete-plugin.ts

```diff
@@ -0,0 +1,243 @@
+import {
+  DeletePlugin,
+  MagicSpan,
+} from "@lblod/ember-rdfa-editor/editor/input-handlers/delete-handler";
+import {
+  Manipulation,
+  ManipulationGuidance,
+  RemoveEmptyTextNodeManipulation,
+  MoveCursorAfterElementManipulation,
+  Editor,
+  RemoveEmptyElementManipulation,
+} from "@lblod/ember-rdfa-editor/editor/input-handlers/manipulation";
+import { runInDebug } from "@ember/debug";
+import {
+  findLastLi,
+  tagName,
+  getWindowSelection,
+} from "@lblod/ember-rdfa-editor/utils/dom-helpers";
+import {
+  moveCaret,
+  stringToVisibleText,
+  hasVisibleChildren,
+} from "@lblod/ember-rdfa-editor/editor/utils";
+import { RawEditor } from "@lblod/ember-rdfa-editor/editor/raw-editor";
+
+function debug(message: String, object: Object | null = null): void {
+  runInDebug(() => {
+    console.debug(`list delete plugin: ${message}`, object);
+  });
+}
+
+/**
+ * This plugin provides sensible behaviour for delete in lists.
+ * @class ListDeletePlugin
+ * @module plugin/lists
+ */
+export default class ListDeletePlugin implements DeletePlugin {
+  label = "delete plugin for handling lists";
+  hasChanged = false;
+
+  guidanceForManipulation(
+    manipulation: Manipulation
+  ): ManipulationGuidance | null {
+    if (manipulation.type == "moveCursorAfterElement") {
+      return this.guidanceForMoveCursorAfterElement(manipulation);
+    } else if (manipulation.type == "removeEmptyElement") {
+      return this.guidanceForRemoveEmptyElement(manipulation);
+    } else if (manipulation.type == "removeEmptyTextNode") {
+      return this.guidanceForRemoveEmptyTextNode(manipulation);
+    }
+    return null;
+  }
+
+  detectChange(): boolean {
+    if (this.hasChanged) {
+      this.hasChanged = false;
```

Comment:
this is the first time I've seen state in a plugin, not necessarily an issue but we have to make sure the calling code takes this into account.

Reply:
yeah I don't like it, but man the change detection was driving me nuts
it's much easier to detect the change within the function thats doing the changes
Definitely to be improved though, this kind of implicit messaging is asking for trouble